sklearn/ensemble/_weight_boosting.py

```diff
@@ -607,8 +616,10 @@ def _boost_discrete(self, iboost, X, y, sample_weight, random_state):
         # Only boost the weights if I will fit again
         if not iboost == self.n_estimators - 1:
             # Only boost positive weights
-            sample_weight *= np.exp(estimator_weight * incorrect *
-                                    (sample_weight > 0))
+            sample_weight = np.exp(np.log(sample_weight) +
+                                   estimator_weight * incorrect *
+                                   ((sample_weight > 0) |
+                                    (estimator_weight < 0)))
```

Comment:
What was the reason behind `estimator_weight < 0`?

Reply:
Yes, this looks like a backward incompatible change to me. I reverted this change.